pandas/core/algorithms.py

```diff
@@ -123,13 +123,8 @@ def _ensure_data(
         elif is_float_dtype(values) or is_float_dtype(dtype):
             return ensure_float64(values), np.dtype("float64")
         elif is_complex_dtype(values) or is_complex_dtype(dtype):
-
-            # ignore the fact that we are casting to float
-            # which discards complex parts
-            with catch_warnings():
-                simplefilter("ignore", np.ComplexWarning)
-                values = ensure_float64(values)
-            return values, np.dtype("float64")
+            # Complex dtype is not supported coerce to object
+            return ensure_object(values), np.dtype("complex64")
```

Comment:
Not sure whether it should be `np.dtype("object")` or `np.dtype("complex128")`, but "complex64", i.e. `real` and `imag` are float32 (instead of float64), seems to be wrong.

Reply:
we now have complex hashtables, so casting to object should be unnecessary?

Reply:
Updated